packages/typescript-estree/src/parseSettings/resolveProjectList.ts

```diff
@@ -93,16 +93,23 @@ export function resolveProjectList(
   const nonGlobProjects = sanitizedProjects.filter(project => !isGlob(project));
   const globProjects = sanitizedProjects.filter(project => isGlob(project));
 
+  let globProjectPaths: string[] = [];
+
+  if (globProjects.length > 0) {
+    // Although fast-glob supports multiple patterns, fast-glob returns arbitrary order of results
+    // to improve performance. To ensure the order is correct, we need to call fast-glob for each pattern
```

Comment:
OOC, does the order matter?

Reply:
There is a unit test case against the order, so yes, it does matter.

Reply:
But why does the order matter? Shouldn't the unit test just check that the needed files are linted?

Reply:
Folks sometimes do things like `['packages/*/tsconfig.json', 'tsconfig.json']` - and/or entries like `'tsconfig.eslint.jsom'`. The order intent is to first try the lower-level TSConfigs first.

Reply:
Gotcha, thanks!